### PR TITLE
perf: base-chart should not create spec transformer

### PR DESCRIPTION
**Pull request by xile611**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [x] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/chart/box-plot/box-plot-transformer.ts [hunk 1]
```diff
@@ -8,19 +8,18 @@ export class BoxPlotChartSpecTransformer<
...
+      'boxPlot',
+      'minField',
+      'maxField',
+      'q1Field',
+      'medianField',
+      'q3Field',
+      'outliersField',
+      'outliersStyle'
+    ]);
+    (seriesSpec.direction = spec.direction ?? Direction.vertical),
```
Comment (xiaoluoHe) [inline packages/vchart/src/chart/box-plot/box-plot-transformer.ts:21]: 这里还是分号隔开两句吧

File: packages/vchart/src/chart/gauge/gauge-transformer.ts [hunk 2]
```diff
@@ -12,21 +12,17 @@ export class GaugeChartSpecTransformer<
...
     const series = super._getDefaultSeriesSpec(spec);
-    return {
-      ...series,
-      radiusField: spec.radiusField,
 
-      pin: spec.pin,
-      pinBackground: spec.pinBackground,
-      pointer: spec.pointer
-    };
+    series.radiusField = spec.radiusField;
```
Comment (xiaoluoHe) [inline packages/vchart/src/chart/gauge/gauge-transformer.ts:16]: 这里为什么不直接传参给 super._getDefaultSeriesSpec(spec, ["pin", .....])

File: packages/vchart/src/chart/progress/circular/circular-progress-transformer.ts [hunk 3]
```diff
@@ -7,17 +7,13 @@ export class CircularProgressChartSpecTransformer<
...
 
-      cornerRadius: spec.cornerRadius ?? 0,
-      roundCap: spec.roundCap ?? false,
-
-      progress: spec.progress,
-      track: spec.track,
-
-      tickMask: spec.tickMask
-    };
+    series.progress = spec.progress;
```
Comment (xiaoluoHe) [inline packages/vchart/src/chart/progress/circular/circular-progress-transformer.ts:11]: 同上

File: packages/vchart/src/chart/radar/radar-transformer.ts [hunk 4]
```diff
@@ -7,25 +7,23 @@ export class RadarChartSpecTransformer<
...
-        spec.area
-      ),
-      seriesMark: spec.seriesMark ?? 'area',
-      activePoint: spec.activePoint,
-      pointDis: spec.pointDis,
-      pointDisMul: spec.pointDisMul,
-      markOverlap: spec.markOverlap
-    };
+    const series = super._getDefaultSeriesSpec(spec);
+    series.line = spec.line;
```
Comment (xiaoluoHe) [inline packages/vchart/src/chart/radar/radar-transformer.ts:11]: 同上

File: packages/vchart/src/chart/gauge/gauge-transformer.ts (hunk 2 above)
Comment (xile611) [reply packages/vchart/src/chart/gauge/gauge-transformer.ts:16]: 因为不想每一层都封装了

**Review by xiaoluoHe — APPROVED** [review]

**Pull request merged**